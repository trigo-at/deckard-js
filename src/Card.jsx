import styled from 'styled-components';
import PropTypes from 'prop-types';
import {borderRadius, boxShadow} from 'styled-system';

const Card = styled.div`
    ${boxShadow} ${borderRadius};
`;

Card.propTypes = {
    ...boxShadow.propTypes,
    ...borderRadius.propTypes,
    borderColor: PropTypes.string,
    borderWidth: PropTypes.oneOf([1, 2]),
};

Card.defaultProps = {
    borderRadius: 1,
    boxShadow: 0,
};

Card.displayName = 'Card';

export default Card;
