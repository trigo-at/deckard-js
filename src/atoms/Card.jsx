import styled from 'styled-components';
import PropTypes from 'prop-types';
import {borderRadius} from 'styled-system';
import Box from './Box';
import theme from '../theme';

const boxShadow = props => {
    const boxShadows = {
        sm: {
            'box-shadow': props.theme.boxShadows.default,
        },
        md: {
            'box-shadow': props.theme.boxShadows.md,
        },
        lg: {
            'box-shadow': props.theme.boxShadows.lg,
        },
        inner: {
            'box-shadow': props.theme.boxShadows.inner,
        },
    };
    return (
        boxShadows[props.boxShadowSize] || {
            'box-shadow': props.theme.boxShadows.default,
        }
    );
};

const boxBorder = props => ({
    border: `${props.borderWidth}px solid ${props.theme.colors[props.borderColor || 'snow']}`,
});

const Card = styled(Box)`
  ${boxShadow} ${boxBorder} ${borderRadius};
`;

Card.propTypes = {
    boxShadowSize: PropTypes.oneOf(['sm', 'md', 'lg', 'inner']),
    borderColor: PropTypes.string,
    ...borderRadius.propTypes,
    borderWidth: PropTypes.oneOf([1, 2]),
};

Card.defaultProps = {
    borderColor: 'borderGray',
    borderRadius: 1,
    borderWidth: 1,
    theme,
};

Card.displayName = 'Card';

export default Card;
