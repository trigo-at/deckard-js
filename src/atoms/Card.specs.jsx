import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';

test('Card matches snapshot', () => {
    const component = renderer.create(
        <Card>
            <span>Card</span>
            <span>Card</span>
            <span>Card</span>
        </Card>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
