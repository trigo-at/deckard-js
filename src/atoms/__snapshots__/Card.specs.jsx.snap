// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
.c1 {
  box-sizing: border-box;
}

.c0 {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.10);
  border: 1px solid #b8c2cc;
  border-radius: 1px;
}

<div
  className="c0 c1"
>
  <span>
    Card
  </span>
  <span>
    Card
  </span>
  <span>
    Card
  </span>
</div>
`;
